tracer/src/Datadog.Trace/IAST/Propagation/StringModuleImpl.cs

```diff
@@ -0,0 +1,303 @@
+// <copyright file="StringModuleImpl.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Datadog.Trace.Iast.Dataflow;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Iast.Propagation;
+
+internal static class StringModuleImpl
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(StringModuleImpl));
+
+    internal static bool CanBeTainted(string? txt)
+    {
+        return txt != null && txt.Length > 0;
+    }
+
+    internal static TaintedObject? GetTainted(TaintedObjects taintedObjects, object? value)
+    {
+        return value == null ? null : taintedObjects.Get(value);
+    }
+
+    /// <summary> Mostly used overload </summary>
+    /// <param name="left"> Param 1 </param>
+    /// <param name="right"> Param 2</param>
+    /// <param name="result"> Result </param>
+    /// <param name="filter"> Literal filter </param>
+    /// <returns> resiñt </returns>
+    public static string OnStringConcat(string left, string right, string result, AspectFilter filter = AspectFilter.None)
+    {
+        try
+        {
+            if (string.IsNullOrEmpty(left) || string.IsNullOrEmpty(right))
+            {
+                return result;
+            }
+
+            if (!CanBeTainted(result) || (!CanBeTainted(left) && !CanBeTainted(right)))
```

Comment:
minor: if might be more readable `CantBeTainted` then `!CanBeTainted`.

Reply:
Done!